### DISCUSSION TITLE
Image import error in angular

### DISCUSSION THREAD

**Discussion by shree-2802** [community/community#130693] [Programming Help]
### Body

Hello everyone, I'm new to angular and I'm facing an issue in importing an image in angular 

Code:import { Component, OnInit } from '@angular/core';
import { APP_CONSTANTS } from '../../constants/appConstants';
import movies from "../../assets/images/movies.png";

@Component({
  selector: 'app-login',
  standalone: true,
  imports: [],
  templateUrl: './login.component.html',
  styleUrl: './login.component.scss'
})
export class LoginComponent {
  loginContent = APP_CONSTANTS.loginContent;
  image=movies;
}

Error: No loader is configured for ".png" files: src/app/assets/images/movies.png

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by HemalathaT2** [comment]
In your angular.json file, add the configuration:
{
  "projects": {
    "your-app": {
     ...
      "architect": {
        "build": {
         ...
          "options": {
           ...
            "assets": [
              "src/assets"
            ]
          }
        }
      }
    }
  }
}

In your template, use the image variable like this:
<img [src]="image" alt="Movies">

Make sure to adjust the image path according to your project structure.

Have a good day!

**Reply by shree-2802** [reply]
I've imported the image but I get this error, "No loader is configured for ".png" files: src/app/assets/images/movies.png"

**Comment by HemalathaT2** [comment]
"assets": [
  "src/favicon.ico",
  "src/assets"
],
"styles": [],
"scripts": [],
"module": {
  "rules": [
    {
      "test": /\.(png|jpg|gif|jpeg)$/,
      "loader": "file-loader",
      "options": {
        "name": "[name].[ext]"
      }
    }
  ]
}
1. Open your angular.json file.
2. Add the following configuration to the projects > your-project-name > architect > build > options section:

**Comment by Mori-Takahashi** [comment]
Hi which Angular version are you using?

**Comment by ricardious** [comment]
Perdón por no escribir en inglés pero la escritura no se me da muy bien.
Parece que el problema está relacionado con la forma en que Angular maneja las importaciones de imágenes

## Solución

1. **Instalar file-loader**

   ```bash
   npm install --save-dev file-loader
   ```

2. **Configurar Webpack**
Tienes que hacer algo como esto

   ```javascript
   const path = require('path');

   module.exports = {
       module: {
           rules: [
               {
                   test: /\.(png|jpe?g|gif|svg)$/i,
                   use: [
                       {
                           loader: 'file-loader',
                           options: {
                               name: '[path][name].[ext]',
                           },
                       },
                   ],
               },
           ],
       },
   };
   ```

3. Asegúrate de que la ruta de importación sea correcta. La ruta debe ser relativa al archivo donde estás importando la imagen. La declaración de importación que tienes parece estar bien, pero verifica que la imagen esté realmente en `src/assets/images/movies.png`.

4. Asegúrate de que el decorador de tu componente esté correctamente definido (parece que hay un error tipográfico en el decorador `@Component`, debería estar en mayúscula):

   ```typescript
   import { Component } from '@angular/core';
   import { APP_CONSTANTS } from '../../constants/appConstants';
   import movies from '../../assets/images/movies.png';

   @Component({
       selector: 'app-login',
       standalone: true,
       imports: [],
       templateUrl: './login.component.html',
       styleUrls: ['./login.component.scss'] // Corrección aquí
   })
   export class LoginComponent {
       loginContent = APP_CONSTANTS.loginContent;
       image = movies;
   }
   ```

5. Después de hacer estos cambios, reinicia tu servidor de desarrollo de Angular
